app/kuma-dp/pkg/dataplane/envoy/envoy.go

```diff
@@ -29,21 +29,24 @@ var (
 
 type BootstrapParams struct {
 	Dataplane       *rest.Resource
-	DNSPort         uint32
-	EmptyDNSPort    uint32
 	EnvoyVersion    EnvoyVersion
 	DynamicMetadata map[string]string
 }
 
-type BootstrapConfigFactoryFunc func(url string, cfg kuma_dp.Config, params BootstrapParams) ([]byte, error)
+type BootstrapConfigGenerator interface {
+	Generate(url string, cfg kuma_dp.Config, params BootstrapParams) ([]byte, error)
```

Comment:
Not all these parameters are intrinsic to the `Generate` operation. We don't need to pass `url` and `cfg` since they can be properties of the generator itself.

See https://github.com/kumahq/kuma/issues/2185